### PR TITLE
Enable activity tracking at password reset

### PR DESCRIPTION
**Pull request by ybelenko**
I'm not really familiar with TypeScript so please check carefully.
I've set `admin: true`, because scope of the token has been already verified few lines above. There is also `permissions` property of `accountability`, but I think password reset is default `allow` permission of any user role by default. Correct me if I'm wrong.

Closes #11857

### CODE REVIEW

File: api/src/services/users.ts [hunk 1]
```diff
@@ -392,6 +392,13 @@ export class UsersService extends ItemsService {
 		const service = new UsersService({
 			knex: this.knex,
 			schema: this.schema,
+			accountability: {
+				role: null,
+				user: user!.id,
+				admin: true,
+				ip: this.accountability!.ip,
+				userAgent: this.accountability!.userAgent,
+			},
```
Comment (rijkvanzanten) [inline api/src/services/users.ts:401]:
```suggestion
			accountability: {
			    ...(this.accountability ?? {}),
			    admin: true, // We need to skip permissions checks for the update call below
			},
```

**Comment by licitdev** [comment]
Allow password updates to show up in the revisions drawer.

### Before
image

### After

image

File: app/src/views/private/components/revisions-drawer-detail/revisions-drawer-updates.vue [hunk 2]
```diff
@@ -61,7 +61,7 @@ export default defineComponent({
 					const currentValue = props.revision.delta[fieldKey];
 					const previousValue = previousRevision.value.data[fieldKey];
 
-					if (isEqual(currentValue, previousValue)) return null;
+					if (isEqual(currentValue, previousValue) && currentValue !== '**********') return null;
```
Comment (rijkvanzanten) [inline app/src/views/private/components/revisions-drawer-detail/revisions-drawer-updates.vue:64]: I don't think we should rely on the arbitrary string value `******` for the conceal check.  It's not unthinkable that the redaction string is going to be configurable in the future, which would cause issues here.

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

File: app/src/views/private/components/revisions-drawer-detail/revisions-drawer-updates.vue (hunk 2 above)
Comment (rijkvanzanten) [reply app/src/views/private/components/revisions-drawer-detail/revisions-drawer-updates.vue:64]: > I think we would still require a check if a string is concealed.

We should pull that information from the field store than, not the returned value 👍🏻

Comment (licitdev) [reply app/src/views/private/components/revisions-drawer-detail/revisions-drawer-updates.vue]: Oh yes! Done. Thanks!!

**Comment by licitdev** [comment]
## Display message when concealed field is updated

image

**Comment by rijkvanzanten** [comment]
@benhaynes Thoughts on that exact message string? 

TL;DR: Fields might be "concealed" (like user password), meaning that even though the value was changed, it went from `*******` to `*******` as far as the app is concerned. We'd still like to show that the `password` field was updated though, hence the notice with a message. Right now it says `Field has been updated but we are not aware if the value has changed` but I'm wondering if there's a better phrasing 🤔

File: app/src/views/private/components/revisions-drawer-detail/revisions-drawer-updates-change.vue [hunk 3]
```diff
@@ -101,6 +110,12 @@ export default defineComponent({
 	}
 }
 
+.updated {
+	color: var(--warning);
+	background-color: var(--warning-alt);
+	border-radius: var(--border-radius) !important;
```
[the remark this answers is not in the document]
Comment (benhaynes) [reply app/src/views/private/components/revisions-drawer-detail/revisions-drawer-updates-change.vue:116]: > Right now it says `Field has been updated but we are not aware if the value has changed` but I'm wondering if there's a better phrasing 🤔

```
This field has been updated but its value is being concealed for security reasons.
```

**Review by rijkvanzanten — APPROVED** [review]
